### PR TITLE
Upgrade PrimeNg-15-lte to PrimeNg 17

### LINKED ISSUE #29606: Upgrade deps with PrimeNG v17
### Parent Issue

https://github.com/dotCMS/core/issues/29058

### Task

As a front-end developer, I need to update PrimeNG from version 15 to version 17 to leverage the latest features, improvements, and bug fixes provided by the newer version.

### PR DESCRIPTION
**Pull request by oidacra**
### Parent Issue
https://github.com/dotCMS/core/issues/29606

### Additonal Info
This PR is not for the master branch, but is for the **epic-primeng-upgrade** branch to break this epic into smaller PRs.

### Proposed Changes
* Upgrade PrimeNg-15-lte to PrimeNg 17

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts [hunk 1]
```diff
@@ -22,34 +22,33 @@ import { DotMessageService } from '@dotcms/data-access';
 import { DotPagesState, DotPageStore } from '../dot-pages-store/dot-pages.store';
 import { DotActionsMenuEventParams } from '../dot-pages.component';
 
+interface CustomContextMenu extends PrimeNGContextMenu {
+    show(event?: unknown): void;
+}
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:27]: If this is a pure primeng component how come we can just change the type?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts [hunk 2]
```diff
@@ -22,34 +22,33 @@ import { DotMessageService } from '@dotcms/data-access';
...
-
     dotStateLabels = {
         archived: this.dotMessageService.get('Archived'),
         published: this.dotMessageService.get('Published'),
         revision: this.dotMessageService.get('Revision'),
         draft: this.dotMessageService.get('Draft')
     };
+    private domIdMenuAttached = '';
+    private destroy$ = new Subject<boolean>();
+    private scrollElement?: HTMLElement;
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:49]:
```suggestion
    #domIdMenuAttached = '';
    #destroy$ = new Subject<boolean>();
    #scrollElement?: HTMLElement;
```

File: core-web/apps/dotcms-ui/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts [hunk 3]
```diff
@@ -113,7 +114,12 @@ export class DotPageSelectorComponent implements ControlValueAccessor {
         if (this.searchType === 'site') {
             const site: Site = <Site>item.payload;
             this.currentHost = site;
-            this.autoComplete.completeMethod.emit({ query: `//${site.hostname}/` });
+            const event: AutoCompleteCompleteEvent = {
+                query: `//${site.hostname}/`,
+                originalEvent: {} as Event
+            };
+            // this.autoComplete.completeMethod.emit({ query: `//${site.hostname}/` });
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts:121]: Prob clean up, just adding it here so is not forgotted.

File: core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.ts [hunk 4]
```diff
@@ -77,14 +77,11 @@ export class DotListingDataTableComponent implements OnInit {
     isContentFiltered = false;
     dateColumns: DataTableColumn[];
     loading = true;
-    contextMenuItems: MenuItem[];
+    contextMenuItems: CustomMenuItem[];
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.ts:80]: Why we need a custom if we passing this to a primeng component?

File: core-web/apps/dotcms-ui/src/app/view/components/dot-portlet-base/components/dot-portlet-toolbar/dot-portlet-toolbar.component.ts [hunk 5]
```diff
@@ -37,7 +37,7 @@ export class DotPortletToolbarComponent {
      * @param {MouseEvent} $event
      * @memberof DotPortletToolbarComponent
      */
-    onPrimaryClick($event: MouseEvent): void {
+    onPrimaryClick($event: unknown): void {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-portlet-base/components/dot-portlet-toolbar/dot-portlet-toolbar.component.ts:40]: Why?

File: core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts [hunk 6]
```diff
@@ -1,6 +1,15 @@
-import { MenuItem } from 'primeng/api';
+import { MenuItem, MenuItemCommandEvent } from 'primeng/api';
 
-export interface DotActionMenuItem {
+export interface CustomMenuItem<T = unknown> extends Omit<MenuItem, 'command'> {
+    command?(event?: T): void;
+}
```
Comment (fmontes) [inline core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts:5]: We remove command with the Omit and then added back? we need to change the type in the function?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_autocomplete.scss [hunk 7]
```diff
@@ -90,7 +91,7 @@ p-autocomplete.p-inputwrapper-focus {
             &:hover,
             &:active,
             &.p-focus {
-                border: none;
+                //border: none;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_autocomplete.scss:94]: Remember clean up.

File: core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts (hunk 6 above)
Comment (oidacra) [reply core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts:5]: In a few cases, we call a command without parameter, and the lint fails.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts (hunk 1 above)
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts:27]: Could be a PrimeNG issue

File: core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.stories.ts [hunk 8]
```diff
@@ -20,9 +22,7 @@ import {
 } from '@dotcms/data-access';
 import { DotSpinnerModule } from '@dotcms/ui';
 
-import { DotBlockEditorComponent } from './dot-block-editor.component';
-
-import { BlockEditorModule } from '../../block-editor.module';
+import { DotAiServiceMock } from '../../../../../data-access/src/lib/dot-ai/dot-ai-service.mock';
```
Comment (zJaaal) [inline core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.stories.ts:25]: should we import this as `@dotcms/data-access`

File: core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts [hunk 9]
```diff
@@ -1,6 +1,15 @@
...
+
+export interface DotActionMenuItem<T = unknown> {
     shouldShow?: (x?: Record<string, unknown>) => boolean;
```
Comment (zJaaal) [inline core-web/libs/dotcms-models/src/lib/dot-action-menu-item.model.ts:8]: what is exactly `x`?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss [hunk 10]
```diff
@@ -238,11 +244,12 @@ p-calendar.p-calendar-clearable {
     }
 
     .p-calendar-clear-icon {
-        margin-top: -12px;
-        color: $color-palette-gray-500;
+        margin-top: -7px;
```
Comment (zJaaal) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:247]: This value can change due to the font size update [PR](https://github.com/dotCMS/core/pull/29205)

**Review by zJaaal — CHANGES_REQUESTED** [review]

**Comment by nicobytes** [comment]
Based on the storybook, I find the following issues:

v15             |  v17
:-------------------------:|:-------------------------:
DotCMS/ThemeSelector | DotCMS/ThemeSelector
Screenshot 2024-08-09 at 9 40 15 AM | Screenshot 2024-08-09 at 9 40 36 AM
DotCMS/Menu/Vertical | DotCMS/Forms/Vertical 
Screenshot 2024-08-09 at 9 42 33 AM | Screenshot 2024-08-09 at 9 42 44 AM
DotCMS/Menu/DotCollapseBreadcrumb | DotCMS/Menu/DotCollapseBreadcrumb
Screenshot 2024-08-09 at 9 52 25 AM  | Screenshot 2024-08-09 at 9 52 38 AM
DotCMS/Menu/DotCrumbtrail | DotCMS/Menu/DotCrumbtrail
Screenshot 2024-08-09 at 9 54 19 AM | Screenshot 2024-08-09 at 9 54 28 AM
PrimeNg/Button/SplitButton | PrimeNg/Button/SplitButton
Screenshot 2024-08-09 at 9 56 26 AM | Screenshot 2024-08-09 at 9 56 38 AM
Screenshot 2024-08-09 at 9 57 44 AM  | Screenshot 2024-08-09 at 9 58 01 AM
PrimeNg/Data/Table | PrimeNg/Data/Table
Screenshot 2024-08-09 at 10 00 49 AM | Screenshot 2024-08-09 at 9 59 58 AM
PrimeNg/Data/Tree | PrimeNg/Data/Tree
Screenshot 2024-08-09 at 10 03 36 AM | Screenshot 2024-08-09 at 10 03 22 AM
PrimeNg/Form| PrimeNg/Form
Screenshot 2024-08-09 at 10 11 49 AM | Not working
Screenshot 2024-08-09 at 10 13 41 AM  |   Screenshot 2024-08-09 at 10 14 21 AM
Screenshot 2024-08-09 at 10 16 59 AM | Not working
Screenshot 2024-08-09 at 10 19 00 AM | Screenshot 2024-08-09 at 10 19 12 AM
Screenshot 2024-08-09 at 10 20 35 AM | Screenshot 2024-08-09 at 10 20 52 AM |
Screenshot 2024-08-09 at 10 22 29 AM | Screenshot 2024-08-09 at 10 22 44 AM
PrimeNg/Overlay | PrimeNg/Overlay
Screenshot 2024-08-09 at 10 29 50 AM  |  Screenshot 2024-08-09 at 10 30 05 AM
Screenshot 2024-08-09 at 10 31 00 AM | Screenshot 2024-08-09 at 10 31 08 AM

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts (hunk 2 above)
Comment (nicobytes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts]: Applied!

File: core-web/apps/dotcms-ui/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts (hunk 3 above)
Comment (nicobytes) [reply core-web/apps/dotcms-ui/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts:121]: Fxied!

Here's a breaking change: now `(onSelect)` doesn't send the selected item directly, instead it sends an obj with 2 values: originalEvent and value as selected item.

Screenshot 2024-08-16 at 9 50 09 AM

File: core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.ts (hunk 4 above)
Comment (nicobytes) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.ts:80]: Fixed!

File: core-web/apps/dotcms-ui/src/app/view/components/dot-portlet-base/components/dot-portlet-toolbar/dot-portlet-toolbar.component.ts (hunk 5 above)
Comment (nicobytes) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-portlet-base/components/dot-portlet-toolbar/dot-portlet-toolbar.component.ts:40]: Fixed!

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_autocomplete.scss (hunk 7 above)
Comment (nicobytes) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_autocomplete.scss:94]: Fixed!

File: core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.stories.ts (hunk 8 above)
Comment (nicobytes) [reply core-web/libs/block-editor/src/lib/components/dot-block-editor/dot-block-editor.component.stories.ts]: Fixed!

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss (hunk 10 above)
Comment (nicobytes) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/form/_calendar.scss:247]: According to the storybook, it works fine!

Screenshot 2024-08-16 at 10 11 46 AM

**Review by nicobytes — APPROVED** [review]

**Pull request merged**